pkg/cmd/experimental/syncgroups/cli/prunegroups.go

```diff
@@ -0,0 +1,276 @@
+	options := NewPruneGroupsOptions()
+	options.Out = out
+
+	whitelistFile := ""
+	blacklistFile := ""
+	configFile := ""
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s [WHITELIST] [--whitelist=WHITELIST-FILE] [--blacklist=BLACKLIST-FILE] --sync-config=CONFIG-SOURCE", name),
+		Short:   "Prune OpenShift groups referencing records from an external provider.",
+		Long:    pruneGroupsLong,
+		Example: fmt.Sprintf(pruneGroupsExamples, fullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Complete(whitelistFile, blacklistFile, configFile, args, f); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := options.Validate(); err != nil {
+				cmdutil.CheckErr(cmdutil.UsageError(c, err.Error()))
+			}
+
+			err := options.Run(c, f)
+			if err != nil {
+				if aggregate, ok := err.(kerrs.Aggregate); ok {
+					for _, err := range aggregate.Errors() {
+						fmt.Printf("%s\n", err)
+					}
+					os.Exit(1)
+				}
+			}
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmd.Flags().StringVar(&whitelistFile, "whitelist", whitelistFile, "path to the group whitelist file")
+	cmd.Flags().StringVar(&blacklistFile, "blacklist", whitelistFile, "path to the group blacklist file")
+	// TODO enable this we're able to support string slice elements that have commas
+	// cmd.Flags().StringSliceVar(&options.Blacklist, "blacklist-group", options.Blacklist, "group to blacklist")
+	cmd.Flags().StringVar(&configFile, "sync-config", configFile, "path to the sync config")
+	cmd.Flags().BoolVar(&options.Confirm, "confirm", false, "if true, modify OpenShift groups; if false, display groups")
+	cmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().Lookup("output").DefValue = "yaml"
+	cmd.Flags().Lookup("output").Value.Set("yaml")
+
+	return cmd
+}
+
+func (o *PruneGroupsOptions) Complete(whitelistFile, blacklistFile, configFile string, args []string, f *clientcmd.Factory) error {
+	// if args are given, they are OpenShift Group names forming a whitelist
+	if len(args) > 0 {
+		o.Whitelist = append(o.Whitelist, args...)
+	}
+
+	// unpack whitelist file from source
+	if len(whitelistFile) != 0 {
+		whitelistData, err := readLines(whitelistFile)
+		if err != nil {
+			return err
+		}
+		o.Whitelist = append(o.Whitelist, whitelistData...)
+	}
+
+	// unpack blacklist file from source
+	if len(blacklistFile) != 0 {
+		blacklistData, err := readLines(blacklistFile)
+		if err != nil {
+			return err
+		}
+		o.Blacklist = append(o.Blacklist, blacklistData...)
+	}
+
+	yamlConfig, err := ioutil.ReadFile(configFile)
+	if err != nil {
+		return fmt.Errorf("could not read file %s: %v", configFile, err)
+	}
+	jsonConfig, err := kyaml.ToJSON(yamlConfig)
+	if err != nil {
+		return fmt.Errorf("could not parse file %s: %v", configFile, err)
+	}
+	if err := configapilatest.Codec.DecodeInto(jsonConfig, &o.Config); err != nil {
+		return err
+	}
+
+	if f != nil {
```

Comment:
> This is inherited code from oc secrets new so I am not sure why, I think you wrote that code? We can remove the check if it can't be nil.

Remove it and see if you still work.